x/ecocredit/server/core/query_params.go

```diff
@@ -4,17 +4,64 @@ import (
 	"context"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
+	ecocreditv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types"
 	"github.com/regen-network/regen-ledger/x/ecocredit/core"
 )
 
 // Params queries the ecocredit module parameters.
-// TODO: remove params https://github.com/regen-network/regen-ledger/issues/729
-// Currently this is an ugly hack that grabs v1alpha types and converts them into v1beta types.
-// will be gone with #729.
 func (k Keeper) Params(ctx context.Context, _ *core.QueryParamsRequest) (*core.QueryParamsResponse, error) {
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	var params core.Params
-	k.paramsKeeper.GetParamSet(sdkCtx, &params)
-	return &core.QueryParamsResponse{Params: &params}, nil
+
+	allowlistEnabled, err := k.stateStore.AllowListEnabledTable().Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	itr, err := k.stateStore.AllowedClassCreatorTable().List(ctx, ecocreditv1.AllowedClassCreatorPrimaryKey{})
+	if err != nil {
+		return nil, err
+	}
+	defer itr.Close()
+
+	creators := []string{}
+	for itr.Next() {
+		val, err := itr.Value()
+		if err != nil {
+			return nil, err
+		}
+
+		creators = append(creators, sdk.AccAddress(val.Address).String())
+
+	}
+
+	classFees, err := k.stateStore.ClassFeesTable().Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	classFees1, ok := types.ProtoCoinsToCoins(classFees.Fees)
+	if !ok {
+		return nil, sdkerrors.ErrInvalidCoins.Wrap("class fees")
+	}
+
+	basketFees, err := k.basketStore.BasketFeesTable().Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	basketFees1, ok := types.ProtoCoinsToCoins(basketFees.Fees)
+	if !ok {
+		return nil, sdkerrors.ErrInvalidCoins.Wrap("basket fees")
+	}
+
+	return &core.QueryParamsResponse{
+		Params: &core.Params{
+			AllowedClassCreators: creators,
+			AllowlistEnabled:     allowlistEnabled.Enabled,
+			CreditClassFee:       classFees1,
+			BasketFee:            basketFees1,
+		},
```

Comment:
This is great. We should add individual queries these params but keep this for the time being. We may also want to consider adding the other params here so that this is not misleading, in which case we would need to update the param struct to include the other params but also deprecate it. I'll open a separate issue for tracking.

Reply:
#1402